lib/src/utils.dart

```diff
@@ -38,10 +39,15 @@ extension InternalDateTimeRrule on DateTime {
     int minute = 0,
     int second = 0,
     int millisecond = 0,
+    int microsecond = 0,
     bool isUtc = true,
   }) {
-    final constructor = isUtc ? DateTime.utc : DateTime.new;
-    return constructor(year, month, day, hour, minute, second, millisecond);
+    if (isUtc) {
+      return DateTime.utc(
+          year, month, day, hour, minute, second, millisecond, microsecond);
+    }
+    return DateTime(
+        year, month, day, hour, minute, second, millisecond, microsecond);
```

Comment:
Can you please keep the old structure here?

```suggestion
    final constructor = isUtc ? DateTime.utc : DateTime.new;
    return constructor(
      year,
      month,
      day,
      hour,
      minute,
      second,
      millisecond,
      microsecond,
    );
```